SingularityUI/app/components/common/modalButtons/RunNowModal.jsx

```diff
@@ -35,22 +39,34 @@ class RunNowModal extends Component {
   }
 
   handleRunNow(data) {
+    localStorage.setItem(LOCAL_STORAGE_AFTER_TRIGGER_VALUE, data.afterTrigger);
+    const runId = uuid.v4();
+    data.runId = runId;
```

Comment:
That's the object that is returned by the FormModal. I'd be fine with cloning it first to avoid mutating the function argument.